src/components/dashboard/Claim.js

```diff
@@ -430,27 +432,18 @@ const Claim = props => {
   }, [setLoading, handleFaceVerification, dailyUbi, setDailyUbi, showDialog, showErrorDialog])
 
   // constantly update stats but only for some data
-  const [startPolling, stopPolling] = useInterval(gatherStats, 10000)
+  // const [startPolling, stopPolling] = useInterval(gatherStats, 60000)
 
   useEffect(() => {
-    // poll blockchain only if the app not in background
-    // and the claim timer haven't reached zero
-    if (appState === 'active' && !dailyUbi) {
+    //refresh stats when user comes back to app, timer state has changed or dailyUBI has changed
+    if (appState === 'active') {
       // refresh all stats when returning back to app
       // or dailyUbi changed meaning a new cycle started
-      // and start polling once refreshed
-      gatherStats(true).then(startPolling)
+      gatherStats(true)
     }
 
-    return stopPolling
-  }, [appState, dailyUbi])
-
-  useEffect(() => {
-    // trigger getting stats if reached time to claim, to make sure everything is update since we refresh
-    if (isReachedZero) {
-      gatherStats()
-    }
-  }, [isReachedZero])
+    // return stopPolling
+  }, [appState, isReachedZero, dailyUbi])
```

Comment:
```suggestion
  }, [appState, isReachedZero, dailyUbi])
  
  useEffect(() => {   
     // trigger getting stats if reached time to claim, to make sure everything is update since we refresh
     if (/*!Config.claimPolling && */!isReachedZero) {
       return
     }
     
     gatherStats()
   }, [isReachedZero])
```

Reply:
i think we need to refresh stats & claim amount just after timer reaches 0. 
optionally you could add the env var condition here so it will be executed if claim polling enabled

Reply:
not sure what you mean because that's what it will do when isReachedZero will change to 0

Reply:
yes i see, sorry